flink-core/src/test/java/org/apache/flink/configuration/DelegatingConfigurationTest.java

```diff
@@ -88,4 +89,51 @@ private String typeParamToString(Class<?>[] classes) {
 			assertTrue("Foo method '" + configurationMethod.getName() + "' has not been wrapped correctly in DelegatingConfiguration wrapper", hasMethod);
 		}
 	}
+	
+	@Test
+	public void testDelegationConfigurationWithNullPrefix() {
+
+		Configuration backingConf = new Configuration();
+		backingConf.setValueInternal("test-key", "value");
+
+		DelegatingConfiguration configuration = new DelegatingConfiguration(
+				backingConf, null);
+		Set<String> keySet = configuration.keySet();
+
+		assertTrue(keySet.equals(backingConf.keySet()));
```

Comment:
minor, it's better to use `assertEquals(keySet, backingConf.keySet())`  here I think. Below is the same